tests/blockchain/eth/contracts/test_contracts_upgradeability.py

```diff
@@ -103,7 +103,8 @@ def test_upgradeability(temp_dir_path, token_economics):
                                                       SourceDirs(temp_dir_path)])
 
     # Prepare the blockchain
-    blockchain_interface = BlockchainDeployerInterface(provider_uri='tester://pyevm/2', compiler=solidity_compiler)
+    provider_uri = 'tester://pyevm/2'
+    blockchain_interface = BlockchainDeployerInterface(provider_uri=provider_uri, compiler=solidity_compiler)
     blockchain_interface.connect()
     origin = blockchain_interface.client.accounts[0]
     BlockchainInterfaceFactory.register_interface(interface=blockchain_interface)
```

Comment:
Do you want to use a `try, finally` instead in case there is ever a test failure?

```python
BlockchainInterfaceFactory.register_interface(interface=blockchain_interface)
try:
     ...
finally:
     del BlockchainInterfaceFactory._interfaces[provider_uri]
```

If so, you can remove line 125 as well.

Reply:
Thanks, good suggestion
✔️